src/audio/copier/copier.c

```diff
@@ -567,6 +566,33 @@ static int do_multi_endpoint_module_copy(struct copier_data *cd, struct comp_dev
 			}
 		}
 	}
+	if (!ret) {
+		comp_update_buffer_consume(src_c, processed_data.source_bytes);
+		if (!cd->endpoint_num || cd->bsource_buffer)
+			cd->input_total_data_processed += processed_data.source_bytes;
+	}
```

Comment:
these 5 lines are now executed twice - here and in `do_module_copy()` below?

Reply:
sorry, not sure I understand your comment correctly, maybe you could rephrase? And in general - is this commit incomplete? I don't understand which function is converted to the module adapter API here?

Reply:
any update here ?

Reply:
@lgirdwood this piece of code is not repeated twice anymore.